### ISSUE TITLE
Update activity gem to track actions on changing profiles

### ISSUE THREAD

**Issue by dboyer** [Iridescent-CM/technovation-app#4366]
Would be helpful to track on profiles:

- Who made the change
- What was changed
- When it was changed

Note: How would data changed via impersonation be displayed?

**Comment by shaun-technovation** [comment]
A starting point for this one is on QA. All updates to an account should now be displayed in the participant's page in the admin; it should track and show which changes were made and by who. It will also display older activity, but the older activity will just display something generic, "Profile updated" since we weren't tracking changes before now.

**Comment by dboyer** [comment]
Tested on QA and confirmed:

- Edits listed when editing a mentor's account from the admin page
- Edits listed when editing a student's account from the admin page
- Edits listed when editing a judge's account from the admin page


@shaun-technovation There were a couple specific issues/questions:

- When I impersonated the person and made a profile change, the change was still associated with my admin account. I believe this is correct based on what we had previously discussed but just wanted to confirm that.

- If I go to edit a field, don't make any changes, and still click save, it adds an entry in the activity log that is just "profile updated." I think that's okay but wanted to confirm that was working as expected.

- When logged in as an admin and editing the mentor type for a mentor, it lists the change as just "profile updated." I'm updating it at a page like /admin/participants/1354#/ and am not impersonating. It's the first one in the screenshot.

<img width="1092" alt="Screen Shot 2024-01-16 at 9 17 12 PM" src="https://github.com/Iridescent-CM/technovation-app/assets/516764/a4ff80f1-58fb-4665-a9df-3dfc75e2faaf">

- I updated a mentor's location when logged in as an admin (not impersonating) and the location change was not tracked in the activity log

**Comment by shaun-technovation** [comment]
> * When I impersonated the person and made a profile change, the change was still associated with my admin account. I believe this is correct based on what we had previously discussed but just wanted to confirm that.

Yes, that is the correct behavior, whoever is making the update should be the person linked to that activity. 👍 

> * If I go to edit a field, don't make any changes, and still click save, it adds an entry in the activity log that is just "profile updated." I think that's okay but wanted to confirm that was working as expected.

This is an interesting case when no fields are updated. We could either not track anything or list a generic "Profile updated" status. It's a little trickier to display something like "No changes made" because we have old activity when we weren't tracking changes and we can't distinguish between the two (expect maybe by the date the activity was created).

> * When logged in as an admin and editing the mentor type for a mentor, it lists the change as just "profile updated." I'm updating it at a page like /admin/participants/1354#/ and am not impersonating. It's the first one in the screenshot.
>
> * I updated a mentor's location when logged in as an admin (not impersonating) and the location change was not tracked in the activity log

These two are somewhat related. These two things aren't being tracked at the moment, but we definitely can track them. Basically anything we want tracked has to be explicitly tracked, and right now it's only the account field updates that are being tracked. The mentor type is on the mentor profile, and location is an account field, but it goes through a different update path; for both of them we basically need to hook into their update path and track that update activity.

**Comment by dboyer** [comment]
Tracking location changes is the highest priority for me since we've had issues with location in the past. I split that into a new issue https://github.com/Iridescent-CM/technovation-app/issues/4422
